Firestore/Source/API/FIRDocumentReference.mm

```diff
@@ -91,16 +90,15 @@ - (instancetype)initWithKey:(FSTDocumentKey *)key
 
 @implementation FIRDocumentReference (Internal)
 
-+ (instancetype)referenceWithPath:(FSTResourcePath *)path firestore:(FIRFirestore *)firestore {
-  if (path.length % 2 != 0) {
++ (instancetype)referenceWithPath:(const ResourcePath &)path firestore:(FIRFirestore *)firestore {
+  if (path.size() % 2 != 0) {
     FSTThrowInvalidArgument(
         @"Invalid document reference. Document references must have an even "
-         "number of segments, but %@ has %d",
-        path.canonicalString, path.length);
+         "number of segments, but %s has %d",
```

Comment:
`path.size()` returns `size_t`. The correct format specifier for this is `%zu`, or absent that, `%lu` and cast the value to `unsigned long`. There should be a warning about this.